README.md

```diff
@@ -1,16 +1,532 @@
+tiles, potentially at a mix of different resolution levels. Typically, large
+low-resolution tiles are used on the interior of the region, and small
+high-resolution tiles are used on the boundary. This is called a
+[multi-order coverage (MOC)] map. An example, reproduced from the
+[MOCPy documentation], is shown below.
+
+![An example HEALPix multi-order coverage map](https://cds-astro.github.io/mocpy/_images/plot_SDSS_r.png)
+
+Much like MOCs, LIGO/Virgo/KAGRA gravitational-wave probability sky maps are
+stored as multi-resolution HEALPix data sets, but with a vector of
+floating-point values attached to each tile. A multi-order refinement mesh from
+an example sky map is shown below, reproduced from [Singer & Price (2015)].
+
+![Example multi-order refinement mesh from a gravitational-wave probability sky map](https://emfollow.docs.ligo.org/userguide/_images/healpix-adaptive-mesh.svg)
+
+[Reinecke & Hivon (2015)][multi-order] introduced HEALPix interval sets as an
+alternative encoding of MOCs that enables fast and simple unions,
+intersections, and queries. In an interval set, each HEALPix tile is described
+by the interval of pixel indices at some very high resolution
+_level_<sub>max</sub> that are descendents of that tile. In an interval set, a
+region is encoded as a disjoint collection of such intervals. A tile with a
+NESTED address given by (_level_, _ipix_) may be described as the half-open
+interval
+
+[_ipix_ 4<sup>_level_<sub>max</sub> - _level_</sup>,
+(_ipix_ + 1) 4<sup>_level_<sub>max</sub> - _level_</sup>).
+
+We use _level_<sub>max</sub> = 29 because this is the highest resolution at
+which pixel indices can be stored in a signed 64-bit integer. At this
+resolution, each pixel is scarcely 0.4 milliarcseconds across.
+
+The interval set representation is adventageous because there are simple and
+fast algorithms for interval arithmetic and set operations. Interval analysis
+appears in a suprising variety of scientific contexts from [genomics][pyranges]
+to [gravitational wave data quality][ligo-segments]. Because of the many
+business applications of interval arithmetic, intervals are also supported in
+the [PostgreSQL] database through its [range types].
+
+### Spatial primitives in `healpix_alchemy`
+
+The `healpix_alchemy` package provides two [custom column types] for
+[SQLAlchemy]:
+
+#### `healpix_alchemy.Point`
+
+This class represents a point. A column of this type could store the positions
+of galaxies in a catalog. Under the hood, it is just a `BIGINT`.
+
+Wherever you need to bind a Python value to a `healpix_alchemy.Point`, you may
+provide any one of the following:
+* an instance of [astropy.coordinates.SkyCoord]
+* a sequence of two [astropy.units.Quantity] instances with angle units, which
+  will be interpreted as the right ascension and declination of the point in
+  the [ICRS] frame
+* an integer representing the HEALPix NESTED index of the point at
+  _level_ = _level_<sub>max</sub>
```

Comment:
Is this always `level_29`, or is `level_max` somehow adjustable?

Reply:
You could set `level_max` to whatever value you want. But many practitioners (myself, the authors of the range set paper, the authors of mocpy) all use `level_max=29`.